From 99e1df06e2d5856883237deeee78ad18c17856fe Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Sun, 24 Jan 2021 21:48:00 +0100
Subject: [PATCH v2 1/1] dm: core: describe uclass_root_s

make htmldocs creates a warning:

./include/asm-generic/global_data.h:443:
warning: Function parameter or member 'uclass_root_s'
not described in 'global_data'

Correct the member descriptions.

Cc: Simon Glass <sjg@chromium.org>
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
v2:
	include information provided by Simon
---
 include/asm-generic/global_data.h | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/include/asm-generic/global_data.h b/include/asm-generic/global_data.h
index 19f70393b4..b6f707e97e 100644
--- a/include/asm-generic/global_data.h
+++ b/include/asm-generic/global_data.h
@@ -198,13 +198,21 @@ struct global_data {
 	 */
 	struct udevice *dm_root_f;
 	/**
-	 * @uclass_root: head of core tree
+	 * @uclass_root_s:
+	 * head of core tree when uclasses are not in read-only memory.
+	 *
+	 * When uclasses are in read-only memory, @uclass_root_s is not used and
+	 * @uclass_root points to the root node generated by dtoc.
 	 */
 	struct list_head uclass_root_s;
 	/**
-	 * @uclass_root: pointer to head of core tree, if uclasses are in
-	 * read-only memory and cannot be adjusted to use @uclass_root as a
-	 * list head.
+	 * @uclass_root:
+	 * pointer to head of core tree, if uclasses are in read-only memory and
+	 * cannot be adjusted to use @uclass_root as a list head.
+	 *
+	 * When not in read-only memory, @uclass_root_s is used to hold the
+	 * uclass root, and @uclass_root points to the address of
+	 * @uclass_root_s.
 	 */
 	struct list_head *uclass_root;
 # if CONFIG_IS_ENABLED(OF_PLATDATA)
-- 
2.29.2

